src/hooks/useFormatNumber.mdx

```diff
@@ -0,0 +1,46 @@
+import { Meta, Source } from "@storybook/blocks";
+
+<Meta title="Hooks/useFormatNumber" />
+
+# useFormatNumber
+
+| Hook Version | DS Version   |
+| ------------ | ------------ |
+| Added        | `Prerelease` |
+| Latest       | `Prerelease` |
+
+`useFormatNumber` provides a number with commas and handles a range of numbers.
+The formatting is based on standards from the NYPL Writing Style Guide.
+
+## Usage
+
+This function formats a number with commas and handles a range of numbers.
+
+1. If one number is passed, it formats that number with commas.
+2. If two numbers are passed, it creates a range:
+   - Numbers are formatted with commas.
+   - En dash (`–`) is used for the range.
+3. Allows numbers passed as strings (e.g., "123456").
+
+Examples:
+
+- (4382) -> "4,382"
+- (4276835) -> "4,276,835"
+- (1, 99) -> "1–99"
+- (141, 58) -> "58–141"
+- (100, 102) -> "100–102"
+- (10, 100) -> "10–100"
+- ("200", "100") -> "100–200"
+- ("200XX", "100") -> null
+- ("200XX", "100XX") -> null
+
+<Source
+  code={`
+const copiesSold = formatNumber(81450000);
+const suggestedPriceRange = formatNumber(100, 20000);
+
+<Text>
+  As of June 2024, the Animal Crossing franchise has sold over \${copiesSold}
+  copies worldwide. Suggested retail price: $\${suggestedPriceRange}.
```

Comment:
Sorry, I forgot to add this before I submitted my review. Please remove the `\$`.